starlite/contrib/jwt/middleware.py

```diff
@@ -0,0 +1,164 @@
+    AbstractAuthenticationMiddleware,
+    AuthenticationResult,
+    NotAuthorizedException,
+)
+from starlite.connection import ASGIConnection
+from starlite.contrib.jwt.jwt_token import Token
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from starlite.types import ASGIApp, Scopes
+    from starlite.utils import AsyncCallable
+
+
+class JWTAuthenticationMiddleware(AbstractAuthenticationMiddleware):
+    """JWT Authentication middleware.
+
+    This class provides JWT authentication functionalities.
+    """
+
+    def __init__(
+        self,
+        app: "ASGIApp",
+        algorithm: str,
+        auth_header: str,
+        exclude: Optional[Union[str, List[str]]],
+        exclude_opt_key: str,
+        retrieve_user_handler: "AsyncCallable[[Token, ASGIConnection[Any, Any, Any]], Awaitable[Any]]",
+        scopes: "Scopes",
+        token_secret: str,
+    ):
+        """Check incoming requests for an encoded token in the auth header specified, and if present retrieves the user
+        from persistence using the provided function.
+
+        Args:
+            algorithm: JWT hashing algorithm to use.
+            app: An ASGIApp, this value is the next ASGI handler to call in the middleware stack.
+            auth_header: Request header key from which to retrieve the token. E.g. 'Authorization' or 'X-Api-Key'.
+            exclude: A pattern or list of patterns to skip.
+            exclude_opt_key: An identifier to use on routes to disable authentication for a particular route.
+            retrieve_user_handler: A function that receives an instance of 'Token' and returns a user, which can be
+                any arbitrary value.
+            scopes: ASGI scopes processed by the authentication middleware.
+            token_secret: Secret for decoding the JWT token. This value should be equivalent to the secret used to
+                encode it.
+        """
+        super().__init__(app=app, exclude=exclude, exclude_from_auth_key=exclude_opt_key, scopes=scopes)
+        self.algorithm = algorithm
+        self.auth_header = auth_header
+        self.retrieve_user_handler = retrieve_user_handler
+        self.token_secret = token_secret
+
+    async def authenticate_request(self, connection: "ASGIConnection[Any,Any,Any]") -> AuthenticationResult:
+        """Given an HTTP Connection, parse the JWT api key stored in the header and retrieve the user correlating to the
+        token from the DB.
+
+        Args:
+            connection: An Starlite HTTPConnection instance.
+
+        Returns:
+            AuthenticationResult
+
+        Raises:
+            [NotAuthorizedException][starlite.exceptions.NotAuthorizedException]: If token is invalid or user is not found.
+        """
+        auth_header = connection.headers.get(self.auth_header)
+        if not auth_header:
+            raise NotAuthorizedException("No JWT token found in request header")
+        _, _, encoded_token = auth_header.partition(" ")
```

Comment:
```suggestion
        encoded_token = auth_header.partition(" ")[-1]
```

If you want to assert a certain amount of parts, it should be done explicitly. Else grabbing the last part is simpler and more explicit like this.